### ISSUE TITLE
events randomly stop being relayed

### ISSUE THREAD

**Issue by gazhayes** [scsibug/nostr-rs-relay#23]
I'm witnessing some strange behaviour. I've tried this on Ubuntu and OSX:

After publishing `x` number of events of around 90kb in size (`x` could be any number, sometimes 3, sometimes 30), the relay stops sending any new events created after this point to any subscriptions that were created after this point.

It still responds to new connection requests (websockets opens properly etc) but it doesn't send any events created after `x` (for any pubkey, and from any client). It *does* send them to clients that have an active subscription and are listening at the time the event was published, but it does not send the event to any future subscriptions. In effect, all events sent after `x` act as though they are ephemeral events.

I opened the database directly and the events published after `x` **are** being immediately persisted to the DB, so it's not a database issue.

Killing the relay with ctrl-c and restarting makes things work normally again for a while, both for newly created events but also the events published *after* `x` but *prior to restart* are also sent to subscribers upon request, as expected.

No notices are sent to the client.

I'm not a rustacean, but noticed that there are a lot of macros like `debug!` and `info!`. Are these logged somewhere, or is there a way to print these to the terminal? This would hopefully help me narrow down the problem.

**Comment by scsibug** [comment]
Just make sure you set the RUST_LOG=debug environment variable, and that should help see what is going on.  I will try to investigate as well.  Is there a specific commit hash you are running where you see this behavior? I have a Mac, so duplicating on OS X shouldn't be an issue.

**Comment by gazhayes** [comment]
Ok I'll set that env and see what else I can figure out.

There isn't a specific commit hash, in fact I was running with a fairly old version (pulled around 2 months ago) when I first noticed this behaviour, so I pulled the latest today but has the same behaviour.

**Comment by scsibug** [comment]
What client are you using?  The same code from less than 2 months is running on a couple pretty highly trafficked relays, so I would love to try and duplicate this.

**Comment by scsibug** [comment]
I made a quick client that would send 100KB-sized events in a loop every second; and tried a variety of requests with `websocat` to see if I could duplicate.  For subscriptions & connections opened far after the event was published, it seemed like all events were being received.  I tried this up to several hundred events.  Each new REQ would immediately pull the entire relay's worth of events, and then start showing the new ones as well.

If I understand correctly, you are seeing a new REQ unable to see some or all of the previously published events, and only showing the new ones?

Let me know what you find in your testing, and any specific REQ filter strings, client versions, etc.
